src/test/ui/suggestions/fn-ctor-passed-as-arg-where-it-should-have-been-called.stderr

```diff
@@ -0,0 +1,14 @@
+error[E0277]: the trait bound `fn() -> impl T {foo}: T` is not satisfied
+  --> $DIR/fn-ctor-passed-as-arg-where-it-should-have-been-called.rs:17:5
+   |
+LL | fn bar(f: impl T<O=()>) {}
+   | ----------------------- required by `bar`
+...
+LL |     bar(foo);
+   |     ^^^ the trait `T` is not implemented for `fn() -> impl T {foo}`
+   |
+   = help: it looks like you forgot to use parentheses to call the function: `foo()`
```

Comment:
In case anyone is wondering why this is not a suggestion on `foo`'s span: the error itself is already wrongly highlighting the `bar` function instead of its `foo` argument. That's a longstanding issue, but we have some ideas on how to continue with it.